### PR TITLE
Csd cnn

### PR DESCRIPTION
**Pull request by BalsRadu**
The following features were added:
->ml_tools which is a set of scripts for data modeling, processing, visualization, and model training
-> Created a module for automatic CNN architecture search based on the DARTS algorithm
->created a module controlled hyperparameter tuning using the framework Optuna(can also be used for architecture search)
->trained over 200 models with different parameters and architectures and managed to get 90% accuracy on the test set with a model of only 0.3 MB.
-> Small changes and updates to the repo structure and contents.

### CODE REVIEW

File: environment.yml
Comment (hp-peti) [inline environment.yml:1]: it's a pity this we will lose this information, seems useful/important

File: models/quality_control/quality_control_early_stopping.pt
Comment (hp-peti) [inline models/quality_control/quality_control_early_stopping.pt:1]: are you using git lfs? you should

File: models/state_estimator/epoch=99-val_accuracy=0.9203.ckpt
Comment (hp-peti) [inline models/state_estimator/epoch=99-val_accuracy=0.9203.ckpt:1]: lfs?

**Review by hp-peti — CHANGES_REQUESTED** [review]
consider putting binary files (.png, .pkl, everything above a few MB) under Git LFS control. also please migrate branch history (`git-lfs-migrate`) or do squash type merge

File: models/state_estimator/epoch=99-val_accuracy=0.9203.ckpt
Comment (BalsRadu) [reply models/state_estimator/epoch=99-val_accuracy=0.9203.ckpt:1]: For now, this will have to wait a bit, since i don't have permission to install lfs on the development machine. After that, i will add all the models to the lfs and implement your recommendations. It is still a bit in the air if the plots and figures are needed, they might just be put in the .gitignore. There is also a small possibility of a database for those models but this might now come in the near future.

File: environment.yml
Comment (BalsRadu) [reply environment.yml:1]: It's ok, that was an older version of the environment i use in conda. I updated a newer version in the setup/envs/ and renamed it to colibri.yml. There is also another file for an env just for visualization.

File: models/quality_control/quality_control_early_stopping.pt
Comment (BalsRadu) [reply models/quality_control/quality_control_early_stopping.pt:1]: I will, at the first opportunity.
